src/ngraph/runtime/plaidml/plaidml_pass_concat_elision.cpp

```diff
@@ -0,0 +1,118 @@
+//*****************************************************************************
+// Copyright 2017-2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/runtime/plaidml/plaidml_pass_concat_elision.hpp"
+#include "ngraph/graph_util.hpp"
+#include "ngraph/op/concat.hpp"
+#include "ngraph/op/util/binary_elementwise_arithmetic.hpp"
+#include "ngraph/op/util/unary_elementwise_arithmetic.hpp"
+#include "ngraph/pattern/matcher.hpp"
+#include "ngraph/pattern/op/any.hpp"
+#include "ngraph/pattern/op/any_of.hpp"
+#include "ngraph/pattern/op/label.hpp"
+#include "ngraph/runtime/plaidml/plaidml_ops_replicate.hpp"
+
+ngraph::runtime::plaidml::pass::ConcatElision::ConcatElision()
+{
+    auto concat_op =
+        std::make_shared<pattern::op::Label>(element::i8, Shape{}, [](std::shared_ptr<Node> node) {
+            return dynamic_cast<ngraph::op::Concat*>(node.get()) != nullptr;
```

Comment:
dynamic_cast doesn't seem to work properly on Macs. You could use node->description() instead.

Reply:
I did all of the testing for this on a Mac, and `dynamic_cast<>` seems like something the compiler team would make really sure to get right before release... I've been following the conversation on Slack; I haven't done enough experimenting to be able to say that anyone's wrong, but... this seems to be working fine for me here.